posthog-core/src/index.ts

```diff
@@ -923,13 +927,24 @@ export abstract class PostHogCoreStateless {
   }
 
   async flush(): Promise<any[]> {
-    if (!this.flushPromise) {
-      this.flushPromise = this._flush().finally(() => {
+    // Wait for the current flush operation to finish (regardless of success or failure), then try to flush again.
+    // Use allSettled instead of finally to be defensive around flush throwing errors immediately rather than rejecting.
+    const nextFlushPromise = Promise.allSettled([this.flushPromise]).then(() => {
```

Comment:
What happens when `this.flushPromise` is null/undefined? Does `allSettled` work with that?

Reply:
Yeah it does! I initially handled the null case separately but it turned out that I didn't need to so I just simplified the code